### PR TITLE
fix(runtime): bring back in memory cache

### PR DESCRIPTION
**Pull request by ailisp**
Mostly undo https://github.com/near/nearcore/pull/3310, on disk cache helper functions are kept and mark as allow(unused), once we plan to re-enable on disk cache it should be obvious to refer and undo this PR.

Test Plan
---------
CI

### CODE REVIEW

File: runtime/near-vm-runner/tests/test_error_cases.rs [hunk 1]
```diff
@@ -530,16 +525,16 @@ fn test_external_call_error() {
         );
     });
 }
-#[test]
```
Comment (olonho) [inline runtime/near-vm-runner/tests/test_error_cases.rs:533]: Why this change?

**Review by olonho — CHANGES_REQUESTED** [review]
Think in memory cache shall be addition to on-disk cache, not replacing it. Just the deserialized data must be stored in in-memory cache, so code could be way simpler. Let me sketch it.

**Comment by olonho** [comment]
Suggest we'd better use https://github.com/near/nearcore/pull/3865

File: runtime/near-vm-runner/tests/test_error_cases.rs (hunk 1 above)
Comment (ailisp) [reply runtime/near-vm-runner/tests/test_error_cases.rs:533]: This assert disk cache count from 0 to 1, which would fail due to on disk cache is not utilized

**Comment by ailisp** [comment]
@olonho Sounds good, I'll close this one
